sys/drv/bcm2835_rootdev.c

```diff
@@ -165,43 +77,28 @@ static void rootdev_teardown_intr(device_t *pic, device_t *dev,
   intr_event_remove_handler(irq->r_handler);
 }
 
-/* Read 32 bit pending register located at irq_base + offset and run
- * corresponding handlers. */
-static void bcm2835_intr_handle(bus_space_handle_t irq_base, bus_size_t offset,
-                                intr_event_t **events) {
-  uint32_t pending = bus_space_read_4(rootdev_bus_space, irq_base, offset);
+static void rootdev_intr_handler(ctx_t *ctx, device_t *dev) {
+  rootdev_t *rd = dev->state;
+  intr_event_t **events = rd->intr_event;
+  uint32_t pending = in4(BCM2836_LOCAL_INTC_IRQPENDINGN(0));
 
   while (pending) {
-    int irq = ffs(pending) - 1;
-    /* XXX: some pending bits are shared between BASIC and GPU0/1. */
+    unsigned irq = ffs(pending) - 1;
     if (events[irq])
       intr_event_run_handlers(events[irq]);
     pending &= ~(1 << irq);
```

Comment:
IIRC comment about shared bits is important.

Reply:
Yes but it regards the BCM2835 peripheral interrupt controller, not the BCM2836 interrupt controller. The comment has been moved to `bcm2835_intr.c`.